### PR TITLE
[#129082833] Fulfil the general items on open-source checklist

### PR DESCRIPTION
**Pull request by ferigis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+
+1. Clone the repo
+2. `rebar3 compile`
+
+## Usage
+
+### Erlang Shell
+
+First of all Sheldon is an application and it needs to be started. You can use `sheldon:start/0`.
+Sheldon only have two main methods, `sheldon:check/1` and `sheldon:check/2`. As a user, you just need to use those.
```
Comment (elbrujohalcon) [inline README.md]: Sheldon only **has**

File: src/sheldon.erl [hunk 2]
```diff
@@ -35,10 +35,14 @@ start() ->
   {ok, _} = application:ensure_all_started(sheldon),
   ok.
 
+%% @doc Checks the spelling of the passed iodata() as a parameter.
```
Comment (elbrujohalcon) [inline src/sheldon.erl]: of the iodata() received as a parameter

File: src/sheldon.erl [hunk 3]
```diff
@@ -35,10 +35,14 @@ start() ->
...
+%% @equiv check(Text, sheldon_config:default())
 -spec check(iodata()) -> sheldon_result:result().
 check(Text) ->
   check(Text, sheldon_config:default()).
 
+%% @doc Checks the spelling of the passed iodata() as a parameter.
```
Comment (elbrujohalcon) [inline src/sheldon.erl]: Same as above

File: src/sheldon_dictionary.erl [hunk 4]
```diff
@@ -48,23 +48,28 @@
...
+%% @doc returns a bazinga from the ETS
 -spec get_bazinga(language()) -> string().
 get_bazinga(Lang) ->
   BazingaName = bazinga_name(Lang),
   Keys = ets:tab2list(BazingaName),
   {Bazinga} = lists:nth(rand:uniform(length(Keys)), Keys),
   Bazinga.
 
+%% @doc returns the name of the dictionary givin the language() as a
+%%      parameter
```
Comment (elbrujohalcon) [inline src/sheldon_dictionary.erl]: givin -> given

File: src/sheldon_utils.erl [hunk 5]
```diff
@@ -54,10 +56,13 @@ normalize(Word) ->
   Escaped = escape_chars(Word, CharToScape),
   binary_to_list(Escaped).
 
+%% @doc checks if iodata() can be a number
 -spec is_number(iodata()) -> boolean().
 is_number(Word) ->
   re:run(Word, "^[0-9]*$") =/= nomatch.
 
+%% @doc checks if some string() match in one of the patterns given as
```
Comment (elbrujohalcon) [inline src/sheldon_utils.erl]: Match -> matches

File: README.md [hunk 6]
```diff
@@ -1,2 +1,68 @@
...
+## Installation
+
+1. Clone the repo
+2. `rebar3 compile`
+
+## Usage
+
+### Erlang Shell
+
+First of all Sheldon is an application and it needs to be started. You can use `sheldon:start/0`.
```
Comment (harenson) [inline README.md]: You should create a file to do all this, something like [`./console`](https://github.com/inaka/gadget/blob/master/console) from gadget or check how does the command [`rebar3 shell`](https://www.rebar3.org/docs/configuration) works.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,68 @@
...
+`sheldon:check/1` and `sheldon:check/2` have the same result type, you can see [sheldon_result.erl](https://github.com/inaka/sheldon/blob/master/src/sheldon_result.erl). Sheldon will return the `ok` atom if the check went well else it'll return
+```erlang
+    #{ misspelled_words := [misspelled_word()]
+     , bazinga          := string()
+     }.
+```
+
+## Dependencies
+
+- Erlang/OTP 19+
```
Comment (harenson) [inline README.md]: Does Erlang/OTP 19.0 works or does it need to be 19.0.5?

File: src/sheldon_utils.erl [hunk 8]
```diff
@@ -31,6 +31,8 @@
 %%% API
 %%%===================================================================
 
+%% @doc normalizes the iodata() escaping some chararcters and converting
```
Comment (harenson) [inline src/sheldon_utils.erl]: Typo: `chararcters`

File: src/sheldon_utils.erl [hunk 9]
```diff
@@ -54,10 +56,13 @@ normalize(Word) ->
   Escaped = escape_chars(Word, CharToScape),
   binary_to_list(Escaped).
 
+%% @doc checks if iodata() can be a number
```
Comment (harenson) [inline src/sheldon_utils.erl]: `... can be a number` -> `... is a number`

**Pull request merged**